lib/network_ssl/network_ssl_compat.c

```diff
@@ -0,0 +1,193 @@
+#include <assert.h>
+#include <limits.h>
+#include <stddef.h>
+#include <string.h>
+
+#include <openssl/ssl.h>
+
+#include "network_ssl_compat.h"
+
+#ifdef NETWORK_SSL_COMPAT_TLS_VERSION
+/**
+ * network_ssl_compat_TLS_client_method():
+ * Create a SSL_METHOD.
+ *
+ * COMPATIBILITY: behave like TLS_client_method().
+ */
+const SSL_METHOD *
+network_ssl_compat_TLS_client_method()
+{
+
+	return (SSLv23_client_method());
+}
+#endif
+
+#ifdef NETWORK_SSL_COMPAT_TLS_VERSION
+/**
+ * network_ssl_compat_CTL_set_min_proto_version(ctx, version):
+ * Set the minimum protocol version to TLS 1.2.  ${version} must be
+ * TLS_1_2_VERSION.
+ *
+ * Return 1 on success, 0 on failure.
+ *
+ * COMPATIBILITY: behave like SSL_CTX_set_min_proto_version(), but only supports
+ * TLS 1.2.
+ */
+int
+network_ssl_compat_CTL_set_min_proto_version(SSL_CTX * ctx, int version)
+{
+
+	/* This the only version currently supported in this file. */
+	assert(version == TLS1_2_VERSION);
+
+	/**
+	 * We should already have a TLS 1.2 connection, because that's the
+	 * only one that network_ssl_compat_TLS_client_method() supports.  But
+	 * to be safe, we'll still disable the other options.
+	 *
+	 * Unfortunately the _set_options() function doesn't return success or
+	 * failure; instead, it returns the new bitmask after setting the
+	 * options.  So we need to AND it with the constant to verify that
+	 * it's been set.
+	 */
+	if ((SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv2) & SSL_OP_NO_SSLv2) == 0)
+		return (0);
+	if ((SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv3) & SSL_OP_NO_SSLv3) == 0)
+		return (0);
+	if ((SSL_CTX_set_options(ctx, SSL_OP_NO_TLSv1) & SSL_OP_NO_TLSv1) == 0)
+		return (0);
+	if ((SSL_CTX_set_options(ctx, SSL_OP_NO_TLSv1_1)
```

Comment:
Why are we turning off the options one by one?  It looks like SSL_CTX_set_options takes a bitmask...?